core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapCastInComparison.java

```diff
@@ -198,6 +201,10 @@ private Expression unwrapCast(ComparisonExpression expression)
             Type sourceType = typeAnalyzer.getType(session, types, cast.getExpression());
             Type targetType = typeAnalyzer.getType(session, types, expression.getRight());
 
+            if (!targetType.isOrderable() && !targetType.isComparable()) {
```

Comment:
This should never happen. If the type of the right term of the comparison expression were not orderable or comparable, the ComparisonExpression would be invalid.

Reply:
Right. This is an additional check arguments like other check arguments as safeguards since this should not add any overheads. Also, I think this is helpful to understand the codes. I can change it to throw instead of silently return. Or do you think it's totally superfluous?

Reply:
Yes, it's superfluous. There's an assumption everywhere in the optimizer that the plan that is provided to each rule is well-formed. It's unreasonable to expect that every rule has to validate that structure. Otherwise, we'll end up with these types of checks sprinkled all over the code and we'll end up with code that's much harder to read and follow.